src/bindgen/ir/annotation.rs

```diff
@@ -68,6 +74,36 @@ impl AnnotationSet {
             .collect();
 
         let must_use = attrs.has_attr_word("must_use");
+        let deprecated = if let Some(note) = attrs.attr_name_value_lookup("deprecated") {
```

Comment:
can we factor this out into its own function? That way you can also early-return and tidy it up a little bit.